include/onnxruntime/core/session/environment.h

```diff
@@ -75,7 +78,7 @@ class Environment {
    * Registers an allocator for sharing between multiple sessions.
    * Return an error if an allocator with the same OrtMemoryInfo is already registered.
    */
-  Status RegisterAllocator(AllocatorPtr allocator);
+  Status RegisterAllocator(OrtAllocator* allocator);
```

Comment:
Separating external from internal usage by making the public method take OrtAllocator*. This enables GetSharedAllocator to return any of the OrtAllocator based instances.